mono/metadata/marshal.c

```diff
@@ -4938,183 +4958,130 @@ mono_marshal_set_last_error_windows (int error)
 #endif
 }
 
-void
-ves_icall_System_Runtime_InteropServices_Marshal_copy_to_unmanaged (MonoArray *src, gint32 start_index,
-								    gpointer dest, gint32 length)
+static gsize
+copy_managed_common (MonoArrayHandle managed, gconstpointer native, gint32 start_index,
+		gint32 length, gpointer *managed_addr, guint32 *gchandle, MonoError *error)
 {
-	ERROR_DECL (error);
-	int element_size;
-	void *source_addr;
+	MONO_CHECK_ARG_NULL_HANDLE (managed, 0);
+	MONO_CHECK_ARG_NULL (native, 0);
 
-	MONO_CHECK_ARG_NULL (src,);
-	MONO_CHECK_ARG_NULL (dest,);
+	MonoClass *klass = mono_handle_class (managed);
 
-	if (m_class_get_rank (mono_object_class (&src->obj)) != 1) {
+	// FIXME? move checks to managed
```

Comment:
If possible, it would be nice to move these to managed, yes.

Reply:
Now or later? I suggest later.